src/esy.ml

```diff
@@ -92,6 +92,19 @@ module State = struct
     | Pending
 end
 
+let find_manifest_in_dir dir =
+  let open Promise.Syntax in
+  let esy_file = Filename.concat dir "esy.json" in
+  let package_file = Filename.concat dir "package.json" in
+  let* esy_file_exists = Fs.exists esy_file in
```

Comment:
If you have the `Promise.List.find_map` primitive you could reuse it here:

```ocaml
["package.json"; "esy.json"] |> Promise.List.find_map ~f:(fun ...)
```

It would also save you a disk access when one file is found immediately.